### ISSUE TITLE
ZC1043 false positive: flags Zsh return parameters (`REPLY`, `reply`, `MATCH`, ...)

### ISSUE THREAD

**Issue by afadesigns** [afadesigns/zshellcheck#1367]
### Summary
ZC1043 ("Use `local` for variables in functions") flags assignments to Zsh's special return parameters inside a function — `REPLY`, `reply`, `MATCH`, `MBEGIN`, `MEND`, `match`, `mbegin`, `mend`, `psvar`. These are the documented mechanism by which a function returns a value to its caller, so they are intentionally global. The advice (and the autofix that prepends `local`) silently breaks value-returning functions.

### Reproduction
```zsh
getval() { REPLY=42 }
getval
print -r -- "$REPLY"   # 42
```
ZC1043 flags `REPLY=42` and its autofix produces `local REPLY=42`, after which the caller reads an empty `$REPLY`.

### Why it is wrong
- `man zshparam`: `REPLY` is "reserved by convention to pass string values"; `reply` is its array form; `read`/`select` set `REPLY`; the `=~` operator and the `b`/`m` glob flags set `MATCH`/`match`/`MBEGIN`/... These are the result mechanism, intentionally global.
- Zsh's own shipped functions use the idiom (`vcs_info`, `_user_expand`). 189 corpus files assign these.
- The autofix compiles, passes re-lint, and returns nothing — silent breakage of correct code.

### Expected
Skip the special return parameters in ZC1043: `REPLY reply MATCH MBEGIN MEND match mbegin mend psvar` (case-sensitive). An explicit user `local REPLY` is already honored.

### Environment
- zshellcheck v1.1.0
